### ISSUE TITLE
We want nz-sider to overlay the content instead of pushing the content

### ISSUE THREAD

**Issue by IsaacHub** [NG-ZORRO/ng-zorro-antd#5521]
## What problem does this feature solve?
nz-sider needs to have an option to make it overlay on top of existing content instead of pushing that content, while expanding the nz-sider from collapsed state, just like material design <mat-drawer mode="over">

https://stackblitz.com/edit/angular-keztzw

## What does the proposed API look like?
A directive like **nzBehavior="overlay | push"** is better
``` <nz-sider nzCollapsible nzBehavior="overlay" nzWidth="200px">```

It needs to have black overlay, once you click the overlay it should go back to folded state.

**Comment by IsaacHub** [comment]
It's unbelievable that this simple basic feature is missing in this library.

**Comment by duongle26** [comment]
You can use `NzDrawer` to make this happens

**Comment by PhilippMatzel** [comment]
Is someone working on this issue?

**Comment by PrabagaranSekar** [comment]
Is this issue fixed? if it means please post it here,
if not suggest me any other solution.
Thanks in advance

**Comment by notthewaquar** [comment]
Is there an update for this issue ??
Please let us know, Thank you !!

**Comment by Jhonatanzr** [comment]
Esto seria supremamente util, poder realizarlo sin tener que depender de otros controles adicionales.

**Comment by predefinedx** [comment]
I was also searching for solution to this problem. After some hours of tweaking here and there, i found one.  `position: fixed` and adding zome `z-index` did the trick for me. In my case, i had to also add height calculation, because my header was also fixed, but after some tweaking, `nz-sider` is overlaying content, instead of pushing it away.

At the moment of writing this comment I cannot provide an example in some sandbox, but I certainly will try to find time, to generate an example of how I achieved this.
